From a6f473be2df0c826a74588b009f86a968500b159 Mon Sep 17 00:00:00 2001
From: Rongrong Zhou <rongrong.zhou@amlogic.com>
Date: Fri, 27 Mar 2015 14:50:53 +0800
Subject: [PATCH 5513/5965] PD#104779: amports: prevent VMASTER D to S when
 tsync disabled

should not move from VMASTER D to VMASTER S due to tsync disable

Change-Id: Ic0e172bb412f46b24fe19db2f294d6cc38bd2f43
---
 drivers/amlogic/amports/tsync.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/amlogic/amports/tsync.c b/drivers/amlogic/amports/tsync.c
index a6a753c550fd..2fc895164435 100755
--- a/drivers/amlogic/amports/tsync.c
+++ b/drivers/amlogic/amports/tsync.c
@@ -389,12 +389,13 @@ static int tsync_mode_switch(int mode,unsigned long diff_pts,int jump_pts)
 
 	printk("%c-discontinue,pcr=%d,vpts=%d,apts=%d,diff_pts=%lu,jump_Pts=%d\n",mode,timestamp_pcrscr_get(),timestamp_vpts_get(),timestamp_apts_get(),diff_pts,jump_pts);
 	if (!tsync_enable) {
-        if(tsync_mode != TSYNC_MODE_VMASTER)
+        if (tsync_mode != TSYNC_MODE_VMASTER) {
 			tsync_mode = TSYNC_MODE_VMASTER;
         tsync_av_mode=TSYNC_STATE_S;
 		tsync_av_dynamic_duration_ms=0;
         printk("tsync_enable [%d] \n",tsync_enable);
 		return 0;
+	}
     }
 	if(mode=='T'){/*D/A--> ...*/
 		if(tsync_av_mode==TSYNC_STATE_D){
-- 
2.19.0

